src/index.js

```diff
@@ -1,28 +1,38 @@
-import {Bond, TimeBond, TransformBond as oo7TransformBond, ReactivePromise} from 'oo7';
-import BigNumber from 'bignumber.js';
-import ParityApi from '@parity/api';
-
-import { abiPolyfill, RegistryABI, RegistryExtras, GitHubHintABI, OperationsABI,
-	BadgeRegABI, TokenRegABI, BadgeABI, TokenABI } from './abis.js';
-
-export function asciiToHex(s) {
-	var r = '0x'
-	for (var i = 0; i < s.length; ++i) {
-		r += ('0' + s.charCodeAt(i).toString(16)).substr(-2);
-	}
-	return r;
-}
+// (C) Copyright 2016-2017 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global parity */
+/* eslint-disable no-return-assign */
+/* eslint-disable no-proto */
+
+const BigNumber = require('bignumber.js');
+const oo7 = require('oo7');
+
+const ParityApi = require('@parity/api');
+
+const { abiPolyfill, RegistryABI, RegistryExtras, GitHubHintABI, OperationsABI, BadgeRegABI, TokenRegABI, BadgeABI, TokenABI } = require('./abis');
 
 function defaultProvider () {
   	if (typeof window !== 'undefined' && window.ethereum) {
   		return window.ethereum;
   	}
 
-	if (typeof ParityApi !== 'undefined') {
-  		return new ParityApi.Provider.Http('http://localhost:8545');
+  	if (typeof window !== 'undefined' && window.parent && window.parent.ethereum) {
```

Comment:
It certainly doesn't hurt, but in most cases (for our UI as well), the cross-origin policy is not going to allow this to work, i.e. for out UI the iframe cannot access the parent.